catalog/cataloger.go

```diff
@@ -168,24 +168,17 @@ type ExportConfigurator interface {
 	PutExportConfiguration(repository string, branch string, conf *ExportConfiguration) error
 }
 
+type stateCB func(oldRef string, state CatalogBranchExportStatus) (newState CatalogBranchExportStatus, newMessage *string, err error)
```

Comment:
1.  If it's useful to us and appears in an exported interface, it could be useful for others.
   ```suggestion
   type ExportStateCallback func(oldRef string, state CatalogBranchExportStatus) (newState CatalogBranchExportStatus, newMessage *string, err error)
   ```
2. Please document the function, here or where it is used.